### PR TITLE
FI-502: Generate unit tests for US Core search tests

### PR DESCRIPTION
**Pull request by Jammjammjamm**
This branch generates unit tests for the search tests in the US Core sequences. The one omission is happy path tests for searches with comparators, which will be addressed in a later ticket. I was planning on refactoring the unit test generation in this branch, but this branch is already large and the refactors obscured the actual changes introduced by this branch.

**Submitter:**
- [x] This pull request describes why these changes were made
- [x] Internal ticket for this PR: https://oncprojectracking.healthit.gov/support/browse/FI-502
- [x] Internal ticket links to this PR
- [x] Internal ticket is properly labeled (Community/Program)
- [x] Internal ticket has a justification for its Community/Program label
- [x] Code diff has been reviewed for extraneous/missing code
- [x] Tests are included and test edge cases
- [x] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

**Comment by arscan** [comment]
Looks like most of the CodeClimate issues are pretty straightforward fixes.

File: generator/uscore/us_core_unit_test_generator.rb [hunk 1]
```diff
@@ -26,6 +26,41 @@ def generate(sequence, path, module_name)
...
+        resource_type:,
+        search_params:,
+        is_first_search:,
+        is_fixed_value_search:,
+        has_comparator_tests:,
+        fixed_value_search_param:,
+        class_name:,
+        sequence_name:
+      )
+        return if resource_type == 'Procedure' && search_params.key?('date')
```
Comment (Jammjammjamm) [inline generator/uscore/us_core_unit_test_generator.rb]: To be removed when #364 is merged.

File: generator/uscore/us_core_unit_test_generator.rb [hunk 2]
```diff
@@ -26,6 +26,41 @@ def generate(sequence, path, module_name)
...
+        return if class_name == 'USCore310DiagnosticreportNoteSequence' && test_key == :search_by_patient_category
```
Comment (Jammjammjamm) [inline generator/uscore/us_core_unit_test_generator.rb]: To be removed when #350 or #352 is merged.

**Comment by czh-orz** [comment]
There are some negative tests I can think of that you aren't testing. For example, I don't think it's checking for failure when the value in the resource returned does not match the parameter we searched by. But I'm okay with this stuff can probably be added later.

**Comment by Jammjammjamm** [comment]
Yeah, I think tests for that and for the comparators are complicated enough to warrant their own ticket.

**Pull request merged**